### PR TITLE
Added lot of stuf. Each commit is atomic

### PR DESCRIPTION
**Pull request by lyrixx**
Todo:
- [x] Updated README
- [x] Turn controller as a service
- [ ] Added unit tests
- [x] Move some code to dedicated class
- [ ] Add a comment feature. But this feature could be disable

### CODE REVIEW

File: src/Gitonomy/Browser/Application.php [hunk 1]
```diff
@@ -6,19 +6,53 @@
...
+                if (!$repository instanceof Repository) {
+                    throw new \RuntimeException(sprintf('Value (%s) in $gitonomy[\'repositories\'] is not an instance of Repository in: "%s"', $key, $configFile));
+                }
+                if (is_string($key)) {
+                    $repository->setName($key);
+                }
+            }
+        } elseif (!is_array($this['repositories'])) {
+            throw new \RuntimeException(sprintf('"$gitonomy" should be a array of Repository or a string in "%s"', $configFile));
+        }
```
Comment (lyrixx) [inline src/Gitonomy/Browser/Application.php]: Move that to a dedicated class ?

File: src/Gitonomy/Browser/Application.php [hunk 2]
```diff
@@ -37,6 +71,25 @@ public function __construct(array $repositories)
...
+                if (!isset($app['repositories'][$repository])) {
+                    $app->abort(404, "Repository $repository does not exist");
+                }
+
+                $this['request_context']->setParameter('repository', $repository);
+
+                $app['twig']->addGlobal('repository', $app['repositories'][$repository]);
+            }
+        });
+
```
Comment (lyrixx) [inline src/Gitonomy/Browser/Application.php]: Move that to a dedicated class ?

File: src/Gitonomy/Browser/Application.php [hunk 3]
```diff
@@ -37,6 +71,25 @@ public function __construct(array $repositories)
...
+        $this['controllers']->convert('repository', function ($repository, Request $request) use ($gitonomy) {
+            if (null !== $repository) {
+                return $gitonomy['repositories'][$repository];
+            }
+        });
+
```
Comment (lyrixx) [inline src/Gitonomy/Browser/Application.php]: Move that to a dedicated class ?

File: src/Gitonomy/Browser/Application.php (hunk 1 above)
Comment (alexandresalome) [inline src/Gitonomy/Browser/Application.php]: move it to a private method `resolveRepositories`?

File: composer.json [hunk 4]
```diff
@@ -1,15 +1,18 @@
 {
     "name": "gitonomy/browser",
     "require": {
-        "silex/silex": "dev-master",
-        "symfony/validator": "2.1.*",
-        "symfony/translation": "2.1.*",
-        "symfony/locale": "2.1.*",
-        "symfony/form": "2.1.*",
-        "twig/twig": ">=1.8,<2.0-dev",
-        "symfony/twig-bridge": "2.1.*",
```
Comment (alexandresalome) [inline composer.json:10]: removal of this line breaks my installation, I get "The function "path" does not exist in "repository_list.html.twig" at line 6"

classes from `symfony/twig-bridge` are missing

Comment (lyrixx) [inline composer.json:10]: oups :( Fixed it asap.

File: src/Gitonomy/Browser/Application.php (hunk 2 above)
Comment (alexandresalome) [inline src/Gitonomy/Browser/Application.php]: this must be located with controllers

File: src/Gitonomy/Browser/Application.php (hunk 3 above)
Comment (alexandresalome) [inline src/Gitonomy/Browser/Application.php]: it should stay with actions

**Comment by alexandresalome** [comment]
it's all OK for me, thank you very much for your contribution.

I pushed two commits in a branch for you to cherry-pick them. One allows you to specify a glob as config (`/var/www/*.dev`) and second is the update of README file

https://github.com/gitonomy/browser/commit/ed093398c8819cada4678f054a682c0df995bbd9
https://github.com/gitonomy/browser/commit/40486e2a6ab72765f1e5abb34f9fdf070eab58b4

I'm looking forward to merging this PR.

File: src/Gitonomy/Browser/Application.php (hunk 2 above)
Comment (lyrixx) [inline src/Gitonomy/Browser/Application.php]: I don't think so. I try do be framework agnostic. So it should be move to an event subscriber.Isn't it

File: src/Gitonomy/Browser/Application.php (hunk 3 above)
Comment (lyrixx) [inline src/Gitonomy/Browser/Application.php]: I don't think so. I try do be framework agnostic. So it should be move to an event subscriber.Isn't it (sorry copy / paste)

Comment (alexandresalome) [inline src/Gitonomy/Browser/Application.php]: true

File: composer.json (hunk 4 above)
Comment (lyrixx) [inline composer.json:10]: Fixed.

**Comment by lyrixx** [comment]
@alexandresalome Could you enable travis ?

I will add the comment feature later (like UT). So this PR is good for review / merge

**Pull request merged**